### ISSUE TITLE
Gradle Build Failed With "unable to load class'com.android.build.gradle.internal.NdkHandler'

### ISSUE THREAD

**Issue by mumer92** [bitcoin-wallet/bitcoin-wallet#365]
Hello,

I cloned repository and been trying to build on latest stable version of Android Studio with updated Gradle (2.14.1). I have lated Ndk Installed via SDK Manager.

However I am unable to build the project.

This is the screenshot of error that I am running into. 

![screen shot 2016-12-13 at 11 14 14 am](https://cloud.githubusercontent.com/assets/960241/21129718/56deb9c4-c125-11e6-9605-75cbaaddaca6.png)

Any help would be appreciated. :)

_root build.gradle file_

````
buildscript {
    repositories {
        jcenter()
    }

    dependencies {
        classpath 'com.android.tools.build:gradle:2.2.3'
        classpath "com.android.tools.build:gradle-experimental:0.7.3"
        classpath('fr.avianey.androidsvgdrawable:gradle-plugin:3.0.0') {
            exclude group: 'xerces'
        }
    }
}

allprojects {
   repositories {
       jcenter()
   }
}
````

_gradle-wrapper.properties_

````
#Mon Dec 12 20:34:43 PKT 2016
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
````

_running grade build in terminal_

```
$ gradle build
To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/2.14.1/userguide/gradle_daemon.html.
Download https://jcenter.bintray.com/com/android/tools/build/gradle/2.2.3/gradle-2.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/build/gradle-core/2.2.3/gradle-core-2.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/build/builder/2.2.3/builder-2.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/lint/lint/25.2.3/lint-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/build/gradle-api/2.2.3/gradle-api-2.2.3.pom
Download https://jcenter.bintray.com/com/android/databinding/compilerCommon/2.2.3/compilerCommon-2.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/build/builder-model/2.2.3/builder-model-2.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/build/builder-test-api/2.2.3/builder-test-api-2.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/sdklib/25.2.3/sdklib-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/sdk-common/25.2.3/sdk-common-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/common/25.2.3/common-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/build/manifest-merger/25.2.3/manifest-merger-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/ddms/ddmlib/25.2.3/ddmlib-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/analytics-library/protos/25.2.3/protos-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/analytics-library/shared/25.2.3/shared-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/analytics-library/tracker/25.2.3/tracker-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/lint/lint-checks/25.2.3/lint-checks-25.2.3.pom
Download https://jcenter.bintray.com/com/android/databinding/baseLibrary/2.2.3/baseLibrary-2.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/layoutlib/layoutlib-api/25.2.3/layoutlib-api-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/dvlib/25.2.3/dvlib-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/repository/25.2.3/repository-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/lint/lint-api/25.2.3/lint-api-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/external/com-intellij/uast/145.597.4/uast-145.597.4.pom
Download https://jcenter.bintray.com/com/android/tools/annotations/25.2.3/annotations-25.2.3.pom
Download https://jcenter.bintray.com/com/android/tools/build/gradle/2.2.3/gradle-2.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/build/gradle-core/2.2.3/gradle-core-2.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/build/builder/2.2.3/builder-2.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/lint/lint/25.2.3/lint-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/build/gradle-api/2.2.3/gradle-api-2.2.3.jar
Download https://jcenter.bintray.com/com/android/databinding/compilerCommon/2.2.3/compilerCommon-2.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/build/builder-model/2.2.3/builder-model-2.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/build/builder-test-api/2.2.3/builder-test-api-2.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/sdklib/25.2.3/sdklib-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/sdk-common/25.2.3/sdk-common-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/common/25.2.3/common-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/build/manifest-merger/25.2.3/manifest-merger-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/ddms/ddmlib/25.2.3/ddmlib-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/analytics-library/protos/25.2.3/protos-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/analytics-library/shared/25.2.3/shared-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/analytics-library/tracker/25.2.3/tracker-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/lint/lint-checks/25.2.3/lint-checks-25.2.3.jar
Download https://jcenter.bintray.com/com/android/databinding/baseLibrary/2.2.3/baseLibrary-2.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/layoutlib/layoutlib-api/25.2.3/layoutlib-api-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/dvlib/25.2.3/dvlib-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/repository/25.2.3/repository-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/lint/lint-api/25.2.3/lint-api-25.2.3.jar
Download https://jcenter.bintray.com/com/android/tools/external/com-intellij/uast/145.597.4/uast-145.597.4.jar
Download https://jcenter.bintray.com/com/android/tools/annotations/25.2.3/annotations-25.2.3.jar

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/umer/Workspace/IdeaProjects/bitcoin-wallet/native-scrypt/build.gradle' line: 1

* What went wrong:
A problem occurred evaluating project ':native-scrypt'.
> java.lang.NoClassDefFoundError: com/android/build/gradle/internal/NdkHandler

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1 mins 41.041 secs

```

**Comment by schildbach** [comment]
Did you install the NDK as described in the README?

**Comment by 3824409A28** [comment]
Hi everyone

**Comment by WePesa** [comment]
Feel free to ping me ismail.malik@gmail.com
We are working on a solution as we speak.

WhatsApp: +447960292474 if you prefer to have a quick call chat.

Rgds,

Ish

**Comment by mumer92** [comment]
Yeah I think Android Studio now allows to install NDK from within SDK Manager.

![screen shot 2016-12-13 at 11 50 08 pm](https://cloud.githubusercontent.com/assets/960241/21154112/03c19c42-c18f-11e6-891e-d3fbab7c2977.png)


I have 2 separate installation of NDK
1) NDK installed by SDK Manager
2) NDK downloaded from [https://developer.android.com/ndk/downloads/index.html](https://developer.android.com/ndk/downloads/index.html )

Seems like the problem is by using `gradle version 2.14.1` with `build tools 2.2.3` and `experimental plugin 0.7.3`.

Now I have reverted to original configuration of project, i.e

```
buildscript {
    repositories {
        jcenter()
    }

    dependencies {
     **classpath 'com.android.tools.build:gradle:2.1.2'
        classpath 'com.android.tools.build:gradle-experimental:0.7.0'**
        classpath('fr.avianey.androidsvgdrawable:gradle-plugin:3.0.0') {
            exclude group: 'xerces'
        }
    }
}

allprojects {
   repositories {
       jcenter()
   }
}

```

Seems like problem with new grade version and new experimental plugin.

**Comment by wbchen99** [comment]
did you fix the issue? I got same problem.

**Comment by Gitju** [comment]
@mumer92, @wbchen99, does it work with Android command line tool for you?

**Comment by liuxgchi** [comment]
I also got the same problem !

**Comment by schildbach** [comment]
Fixed by #532.
